### PR TITLE
EZP-29194: Prefix tags with repository id to support multi repo on same proxy

### PR DESCRIPTION
**Pull request by andrerom**
> Improvment: https://jira.ez.no/browse/EZP-29194

Varnish/Fastly is able to keep page cache separated based on host name and Url, however tags
are currenly not seperated. So to avoid cache claring by tags affecting
several installs which is using different databases, we need to prefix
them using repository name _(if set)_.

Review notes:
- Had to set fos cache instance as alias to make sure what we have configured as arguments are taken into account. Also since the class have been the same since 0.5, this does not change much besides cleanup the compiler pass.
- I opted to decorate Purge Client to try to avoid implementations having to be adapted, including Fastly bundle.
- I opted to skip prefix, if no repository is set, this is the default when installing eZ Platform

#### QA
Can be tested by setting up ezplatform.yml in following way:
```yml
    repositories:
        intranet:
            storage:
(...)
    system:
        site_group:
            repository: intranet
```

With that make sure to test that cache is cleared as you would normally expect, and if you want to see under the hood on tags, make sure to configure your ip as debug ip in varnish so you see headers.

### CODE REVIEW

File: src/Handler/TagHandler.php [hunk 1]
```diff
@@ -64,6 +70,17 @@ public function addTagHeaders(Response $response, array $tags)
             );
         }
 
+        // Prefix tags with repository prefix (to be able to support several repositories on one proxy)
+        // But only if repo prefix is set (when not "default", see ctor), & if not already applied to tags
+        if ($this->repoPrefix && strpos($tags[0], $this->repoPrefix) !== 0) {
```
[the remark this answers is not in the document]
Comment (andrerom) [reply src/Handler/TagHandler.php]: In theory yes, but in practice this function should only be executed once, but FOS event listener, which is ran after all events which adds tags to TagHander has been running.

So unsure if I want to complexity the code with that, but I'm open to suggestion.
Was thinking of making this method private, but does not fit so well in patch release.

**Comment by andrerom** [comment]
@vidarl any thought / review from you on this?

**Comment by andrerom** [comment]
> as long as you are 100% sure it does indeed not break Fastly bundle

Confirmed to work as intended disabled, _and_ when enabled on demo bundle:
> Surrogate-Key: intranet_ez-all intranet_content-75 intranet_content-type-38 intranet_location-76 intranet_parent-71 intranet_path-1 intranet_path-2 intranet_path-95 intranet_path-71 intranet_path-76 intranet_location-2 intranet_parent-1 intranet_content-63 intranet_content-type-42 intranet_content-72 intranet_location-73 intranet_path-73 intranet_content-74 intranet_location-75 intranet_path-75 intranet_content-62 intranet_content-type-16 intranet_location-64 intranet_parent-70 intranet_path-70 intranet_path-64 intranet_content-68 intranet_content-type-18 intranet_location-69 intranet_parent-66 intranet_path-66 intranet_path-69 intranet_content-73 intranet_location-74 intranet_path-74 intranet_content-71 intranet_location-72 intranet_path-72 intranet_content-79 intranet_location-80 intranet_path-80 intranet_content-84 intranet_content-type-27 intranet_location-85 intranet_parent-51 intranet_path-43 intranet_path-51 intranet_path-85 intranet_content-90 intranet_location-91 intranet_path-91

File: src/Handler/TagHandler.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (andrerom) [reply src/Handler/TagHandler.php]: Fixed in 8f76661

**Comment by vidarl** [comment]
> Side: 1002 characters, so almost 1k by now (Varnish limit is 7.5k)

Maybe we should check size then, and output debug warning if we reach varnish limit ?

**Comment by andrerom** [comment]
> Maybe we should check size then, and output debug warning if we reach varnish limit ?

Varnish will do that when you reach it.

**Pull request merged**